javatools/src/main/java/org/xvm/runtime/MainContainer.java

```diff
@@ -38,21 +40,66 @@ public MainContainer(Runtime runtime, NativeContainer containerNative, ModuleCon
+                TypeConstant typeString       = pool.typeString();
+                TypeConstant typeStrings      = pool.ensureArrayType(typeString);
+                TypeConstant requiredType     = type;
+
+                if (requiredType.isNullable()) {
+                    // strip nullable from the required type
+                    requiredType = requiredType.removeNullable();
                 }
-                if (type.equals(frame.poolContext().typeString())) {
-                    return xString.makeHandle(sValue);
+
+                if (oValue instanceof String sValue) {
+                    if (requiredType.equals(typeString)) {
+                        // require String and value is String
+                        return xString.makeHandle(sValue);
+                    } else if (requiredType.equals(typeStrings)) {
+                        // require String[] and value is String
+                        String[] asValue = {sValue};
+                        return xString.makeArrayHandle(asValue);
+                    } else if (requiredType.isA(typeDestringable)) {
+                        // require Destringable and value is String
+                        return toDestringable(frame, requiredType, sValue);
+                    }
+                } else if (oValue instanceof String[] asValue && asValue.length > 0) {
+                    if (requiredType.equals(typeStrings)) {
+                        // require String[] and value is String[]
+                        return xString.makeArrayHandle(asValue);
+                    } else if (requiredType.equals(typeString)) {
+                        // require String and value is String[] so return the last element
+                        // from the array
+                        return xString.makeHandle(asValue[asValue.length - 1]);
+                    }
                 }
             }
         }
 
+        // No matching injectable found, so if the required type is nullable return null, otherwise
+        // return an exception
         ObjectHandle hResource = f_parent.getInjectable(frame, sName, type, hOpts);
-        return hResource == null
-                ? new DeferredCallHandle(xException.makeHandle(frame, "Invalid resource: " + sName))
-                : maskInjection(frame, hResource, type);
+        if  (hResource == null) {
+            return type.isNullable()
+                    ? xNullable.NULL
+                    : new DeferredCallHandle(xException.makeHandle(frame, "Invalid resource: " + sName));
+        }
+        return maskInjection(frame, hResource, type);
+    }
+
+    private ObjectHandle toDestringable(Frame frame, TypeConstant type, String sValue) {
+        ConstantPool    pool     = frame.poolContext();
+        TypeComposition clz      = type.ensureClass(frame);
+        ClassTemplate   template = clz.getTemplate();
+        MethodStructure ctor = template.getStructure()
```

Comment:
Why cut in two lines? Unless you wanted to align the equal signs and it didn't fit, but now it's neither